### DISCUSSION TITLE
Time Machine target disk for MacOS 14 Sonoma and up

### DISCUSSION THREAD

**Discussion by terminationshok** [Netatalk/netatalk#993] [Q&A]
I have been running netatalk 3.1.12 on ubuntu 22.04. My afp.conf looks like:

> [Time Machine]
  path = /mnt/raid/TimeMachine
  time machine = yes

When any mac using Sonoma connects, I see: 

<img width="256" alt="Screenshot 2024-05-16 at 3 14 36 PM" src="https://github.com/Netatalk/netatalk/assets/1037899/30ffec4e-1af4-46b6-98e2-4ccbb91f7d23">

It will say "Connecting to backup disk..." forever. I can click "Skip this backup," and then "Back Up Now," and it works.

Macs on MacOS 13 Ventura down through at least Catalina are working fine on my server.

It is my understanding that AFP is deprecated. [Source](https://kb.synology.com/vi-vn/DSM/tutorial/file_protocol_AFP_not_recommended#x_anchor_id66bb327b22) 

Does this project support Time Machine targets for modern Mac systems?

**Comment by rdmark** [comment]
Yes, netatalk supports Time Machine targets for macOS 14. Just tested it on my MBP with 14.4.1 to make sure nothing broke recently:

<img width="477" alt="Screenshot 2024-05-17 at 9 54 18" src="https://github.com/Netatalk/netatalk/assets/2108467/edb98c43-2845-4e7f-bdbd-c9ac338b3551">
<img width="232" alt="Screenshot 2024-05-17 at 9 55 18" src="https://github.com/Netatalk/netatalk/assets/2108467/9d43a9b6-faf0-41f5-a233-b85f878c478d">

To get to the bottom of why it's not working properly for you, can you please grab some debug logs from initialization of netatalk to when the Sonoma client attempt the first backup?

AFP is definitely a "legacy" protocol in macOS. Apple still ships the OS with the `mount_afp` CLI tool so they're keeping it alive for now.

**Reply by terminationshok** [reply]
I get quite a few lines like this: 

> May 16 19:35:31.435557 afpd[775199] {ad_open.c:812} (error:ad): ad_header_read_ea("/mnt/raid/TimeMachine"): invalid metadata EA this is now being treated as a fatal error. if you see this log entry, please file a bug ticket with your upstream vendor and attach the generated core file.

**Reply by terminationshok** [reply]
I could move to ubuntu 24.04, which supports netatalk 3.1.18. Or I can install netatalk 3.1.18 from source.

It might also be time to move my time machine share to samba. It seems that following [this guide](https://netatalk.io/docs/Installing-Netatalk-3-on-Ubuntu) could allow time machine to find the samba hosted share.

**Reply by rdmark** [reply]
Hm, there is something that doesn't add up here. The "invalid metadata EA" error messages were added in 3.1.14 with https://github.com/Netatalk/netatalk/pull/174 so if you're running 3.1.12 you shouldn't be seeing those at all. Can you please double check the version of netatalk you're running, e.g. with `netatalk -V`

**Reply by APCCV** [reply]
Old thread... but this was fixed with... 3.2.1? Works in 3.2.5 and 4.0.2.
